// node-fitbit/src/index.js
"use strict";


/**
 * Fitbit API Index
 *
 */


var FitbitAPI = {};

var ClientWrapper = require('./modules/client-wrapper');

FitbitAPI.setKeys           = ClientWrapper.setKeys;

FitbitAPI.query             = ClientWrapper.query;

FitbitAPI.getAccessToken    = ClientWrapper.getAccessToken;

FitbitAPI.getRequestToken   = ClientWrapper.getRequestToken;


/** Handle rejection error */
process.on('unhandledRejection', function(err){
    console.log(err.toString());
});


FitbitAPI.setKeys({
    CONSUMER_KEY: '3cfe217b7cc898ee72fa1a95f74a6400',
    CONSUMER_SECRET: '8fd45faba3f3c6ce0f9ac9c5e1c29ef6'
});

var value = {
    method: 'GET',
    format: 'json',
    accessToken: '042453b459bce15056dc517bdffe324b',
    accessTokenSecret: '3ccca58601f1821ed2354f4994c81f45',
    data: {
        fat: '7',
        date: '2015-06-05'
    }
};


FitbitAPI
    .query([
        'devices'
       //'user'
       // 'browse-public-activities',
        /**'activities',
        'bodyMeasurements',
        'body-fat',
        'body-weight',
        'badges',
        'weight-goal',
        'fat-goal',
         'public-activities',
        'recent-activity',
        'frequent-activity',
        'activity-daily-goal',
        'activity-weekly-goal',
        'activity-stats'**/
    ], value)
    .then(function(data){
        console.log(data);
    }, function(err){
        console.log(err);
    });
/*
var json = [ { user: { age: 23, avatar: 'https://d6y8zfzc2qfsl.cloudfront.net/F0BD3028-4CAE-2830-4DF9-079AFE42B141_profile_100_square.png', avatar150: 'https://d6y8zfzc2qfsl.cloudfront.net/F0BD3028-4CAE-2830-4DF9-079AFE42B141_profile_150_square.png', country: 'TH', dateOfBirth: '1992-02-15', displayName: 'Taku', distanceUnit: 'METRIC', encodedId: '3GSLVD', foodsLocale: 'en_US', fullName: 'Taku Suzuki', gender: 'MALE', glucoseUnit: 'METRIC', height: 183, heightUnit: 'METRIC', locale: 'ja_JP', memberSince: '2015-05-27', nickname: 'Taku', offsetFromUTCMillis: 32400000, startDayOfWeek: 'SUNDAY', strideLengthRunning: 95.2, strideLengthWalking: 75.9, timezone: 'Asia/Tokyo', topBadges: [], waterUnit: 'en_US', waterUnitName: 'fl oz', weight: 80, weightUnit: 'METRIC' } }, { activities: [], goals: { activeMinutes: 30, caloriesOut: 2996, distance: 8.05, floors: 10, steps: 10000 }, summary: { activeScore: -1, activityCalories: 20, caloriesBMR: 1834, caloriesOut: 1855, distances: [Object], elevation: 0, fairlyActiveMinutes: 0, floors: 0, lightlyActiveMinutes: 6, marginalCalories: 7, sedentaryMinutes: 1434, steps: 76, veryActiveMinutes: 0 } }, { body: { bicep: 0, bmi: 0, calf: 0, chest: 0, fat: 0, forearm: 0, hips: 0, neck: 0, thigh: 0, waist: 0, weight: 0}}, {fat: []}, {weight: [] }, {badges: [] }, {goal: {}}, {goal: {}}, [], [], [{ activityId: 10103705, description: '', mets: -1, name: 'Chest Exercise' } ], { goals: { activeMinutes: 30, caloriesOut: 2996, distance: 8.05, floors: 10, steps: 10000 } }, {goals: {distance: 56.33, floors: 70, steps: 70000 } }, {lifetime: { total: [Object], tracker: [Object] } }, {bp: []}, { foods: [], summary: { calories: 0, carbs: 0, fat: 0, fiber: 0, protein: 0, sodium: 0, water: 0}}, {glucose: [ [Object], [Object], [Object] ] }, {average: [ [Object], [Object], [Object] ], heart: [] }, { sleep: [], summary: { totalMinutesAsleep: 0, totalSleepRecords: 0, totalTimeInBed: 0}}, { 'body-weight': [[Object], [Object], [Object], [Object], [Object], [Object], [Object]]}, {lifetime: { total: [Object], tracker: [Object] } }, {friends: [], hideMeFromLeaderboard: 'false' }, {friends: [] }, {friends: [] }, {meals: [] }, [{ battery: 'High', deviceVersion: 'Surge', id: '33822890', lastSyncTime: '2015-06-04T01:03:17.082', mac: '007A30092EEE', type: 'TRACKER' } ], {apiSubscriptions: [] }, [], [], [], {summary: { water: 0 }, water: [] } ];

json = json.map(function(item){
    var obj = {};
    for(var key in item){
        obj.name = key;
        obj.data = item[key]
    }
    if(!obj.name) return null;
    return obj;
}).filter(function(item){
    return !!item;
})

console.log(json);
*/

module.exports = FitbitAPI;
